### DISCUSSION TITLE
박성은 / 과제 소개

### DISCUSSION THREAD

**Discussion by Elenapark** [wanted-frontend-team2/1-1#6] [본인 과제 소개]
# README
- 구현 내용
- 프로젝트 실행 방법
- 배포 링크 

과제에서 요구한 필수 사항만 작성하였는데, 세부적인 설명이 더 들어가야 좋은 리드미라고 생각합니다. 수정이 필요하겠네요 

# 회원가입,로그인
## Register.js
- 회원가입 유효성 검사로 정규표현식 사용
![스크린샷 2022-12-21 오전 2 39 38](https://user-images.githubusercontent.com/60565155/208731050-9b3ff82f-a237-42aa-b822-5a0d362d9c0a.png)
- email,pw를 하나의 state에서 객체 형태로 관리
![스크린샷 2022-12-21 오전 2 41 09](https://user-images.githubusercontent.com/60565155/208731345-c96d6f97-3759-416d-b806-34503794d367.png)
- email,pw input이 onchange될 때마다 정규표현식 테스트, 유효한 input 값인지 체크
![스크린샷 2022-12-21 오전 2 41 21](https://user-images.githubusercontent.com/60565155/208731381-8e1e8eb7-5ed5-48db-a33a-855903c730e4.png)
- TextField 컴포넌트 내부 helper text를 이용하여 경고문구 표기
![스크린샷 2022-12-21 오전 2 41 43](https://user-images.githubusercontent.com/60565155/208731432-173cab09-ec20-4b0c-ae32-8011f604ff06.png)

## Signin.js
- 유효성 검사 및 helper text 표기 부분을 제외하고는 회원가입 form과 매우 유사합니다. 하나의 form으로 재사용하도록 리팩토링이 필요할 것 같습니다.

# Todo List CRUD
## TodoList.js
- localStorage key를 변수에 저장하여 관리함으로써 유지보수성을 높일수 있습니다.
![스크린샷 2022-12-21 오전 3 02 43](https://user-images.githubusercontent.com/60565155/208735194-54f8b7ea-84d4-454b-8c4e-524904a49b75.png)
- 또한 기존의 localStorage를 그대로 사용하지 않고 customStorage로 개별 모듈로 분리하여 가져왔는데, 이렇게 관리할 경우 브라우저에서 로컬스토리지 데이터에 직접적인 접근 및 수정을 할 경우 발생할 수 있는 에러 상황에 대해 대처할 수 있습니다.
![스크린샷 2022-12-21 오전 3 04 55](https://user-images.githubusercontent.com/60565155/208735559-11106e6f-684f-4946-bcf9-2a5d6291771d.png)

## services/todo_service.js
- service 로직을 분리하여 get,add,update,delete 각각의 fetch함수에서 response.ok가 true인 경우에만 응답 값을 리턴하도록 하였습니다. 
![스크린샷 2022-12-21 오전 2 58 15](https://user-images.githubusercontent.com/60565155/208734409-79bb5f1a-0600-4b23-b2a5-ac2d91cde7a2.png)

# 스타일
- 스타일 점수는 없었지만 빠른 컴포넌트 구성과 스타일 편의를 위해 Material UI를 사용하였습니다.
- Material UI는 자체 기본 theme을 제공하지만 아래와 같이 커스텀하여 사용할 수도 있습니다. 
![스크린샷 2022-12-21 오전 3 11 38](https://user-images.githubusercontent.com/60565155/208736573-e05ba4ee-a212-42c3-aa96-1f6616d57535.png)

# 컨벤션 제안 (제가 실제로 자주 사용했던 것들로만 적었습니다)
- feat 또는 add: 신규 기능 추가
- fix: 버그 수정
- refac: 코드 리팩토링
- style: 스타일링 수정
- remove: console.log 또는 dead code 제거 시

**Comment by ahn0min** [comment]
**customStorage** 에 대해서 궁금한 사람들이 많습니다..🥹
